backend/controller/controller.go

```diff
@@ -187,19 +188,37 @@ func New(ctx context.Context, db *dal.DAL, config Config, runnerScaling scaling.
 	svc.controllerListListeners = append(svc.controllerListListeners, svc.tasks, cronSvc)
+	if _, ok := runnerScaling.(*localscaling.LocalScaling); ok {
+		svc.scheduleDevelopmentTasks()
+	} else {
+		svc.scheduleProductionTasks()
+	}
+	return svc, nil
+}
 
-	svc.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, svc.syncRoutes)
-	svc.tasks.Parallel(backoff.Backoff{Min: time.Second * 3, Max: time.Second * 3}, svc.heartbeatController)
-	svc.tasks.Parallel(backoff.Backoff{Min: time.Second * 5, Max: time.Second * 5}, svc.updateControllersList)
-	svc.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 10}, svc.reapStaleRunners)
-	svc.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 20}, svc.releaseExpiredReservations)
-	svc.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, svc.reconcileDeployments)
-	svc.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, svc.reconcileRunners)
+func (s *Service) scheduleProductionTasks() {
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, s.syncRoutes)
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second * 3, Max: time.Second * 3}, s.heartbeatController)
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second * 5, Max: time.Second * 5}, s.updateControllersList)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 10}, s.reapStaleRunners)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 20}, s.releaseExpiredReservations)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, s.reconcileDeployments)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second * 5}, s.reconcileRunners)
 	// This should only run on one controller, but because dead controllers
 	// might be selected by the hash ring, we have to run it on all controllers.
 	// We should use a DB lock at some point.
-	svc.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second * 20}, svc.reapStaleControllers)
-	return svc, nil
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second * 20}, s.reapStaleControllers)
+}
+
+func (s *Service) scheduleDevelopmentTasks() {
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second}, s.syncRoutes)
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second * 2, Max: time.Second * 2}, s.heartbeatController)
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second}, s.updateControllersList)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second}, s.reapStaleRunners)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second}, s.releaseExpiredReservations)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second}, s.reconcileDeployments)
+	s.tasks.Singleton(backoff.Backoff{Min: time.Second, Max: time.Second}, s.reconcileRunners)
+	s.tasks.Parallel(backoff.Backoff{Min: time.Second, Max: time.Second}, s.reapStaleControllers)
```

Comment:
Nice! This will be much better :)

Is there a way to refactor this so only the `Max` time value is changed? vs. having to duplicate all these tasks? My concern would be that we might add a `task` to dev and forget to add it to prod or visa versa

Reply:
good point - will refactor so we don't need to maintain separate lists